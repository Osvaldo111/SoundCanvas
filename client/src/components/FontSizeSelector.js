import React from "react";
import "../style/fontSizeSelector.css";
import arrowDown from "../images/arrowDown.svg";
import { connect } from "react-redux";

/**
 * Date: 03/06/2019
 * author: Osvaldo Carrillo
 * This class is designed to allow the user to select the font size
 * of the text input
 * This class has to use the method "handleGetFontSize(sizeSelected)"
 * as a property which return the selected color.
 */
class FontSizeSelector extends React.Component {
  constructor(props) {
    super(props);
    this.inputRef = React.createRef();
    this.state = {
      display: false,
      fontSize: "8"
    };
  }

  handleChange = event => {
    this.setState({ fontSize: event.target.value });
  };

  /**
   * Get the font size after the user press the
   * enter button "keycode 13"
   */
  selectFontSizeEnter = event => {
    if (event.keyCode === 13) {
      this.inputRef.current.blur();
      var fontSize = event.target.value;

      if (fontSize > 400) {
        const valueFont = 400;
        this.setState({ fontSize: 400, display: false }, () => {
          this.props.handleGetFontSize(valueFont);
        });
      } else if (fontSize < 8) {
        const valueFont = 8;
        this.setState({ fontSize: 8, display: false }, () => {
          this.props.handleGetFontSize(valueFont);
        });
      } else {
        this.setState({ fontSize: fontSize, display: false }, () => {
          this.props.handleGetFontSize(fontSize);
        });
      }
    }
  };

  /**
   * Get the font size the user selected from the
   * dropdown menu
   */
  selectFontSize = event => {
    if (event.target.tagName === "LI") {
      var fontSize = event.target.getAttribute("data-size");
      this.setState({ fontSize: fontSize }, () => {
        this.props.handleGetFontSize(fontSize);
      });
    }
  };

  /**
   * Open and close the dropdown menu
   */
  displayDropDown = () => {
    const { display } = this.state;
    if (!display) {
      this.setState({ display: true });
    } else {
      this.setState({ display: false });
    }
  };

  render() {
    const { display } = this.state;
    return (
      <div>
        <div>Font Size</div>
        <div className="ftSelConIn" onClick={this.displayDropDown}>
          <input
            type="number"
            value={this.state.fontSize}
            onChange={this.handleChange}
            onKeyUp={this.selectFontSizeEnter}
            ref={this.inputRef}
          />
          <div className="ftSelConInIm">
            <img src={arrowDown} alt="" />
          </div>
        </div>
        <div
          className="ftSelCtnDrpDwn"
          style={{ display: display ? "block" : "none" }}
        >
          <ul onClick={this.selectFontSize}>
            <li data-size="8">8</li>
            <li data-size="6">6</li>
            <li data-size="10">10</li>
            <li data-size="12">12</li>
            <li data-size="14">14</li>
            <li data-size="16">16</li>
            <li data-size="18">18</li>
            <li data-size="21">21</li>
            <li data-size="24">24</li>
            <li data-size="28">28</li>
            <li data-size="32">32</li>
            <li data-size="36">36</li>
            <li data-size="42">42</li>
            <li data-size="48">48</li>
            <li data-size="56">56</li>
          </ul>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {}
const mapDispatchToProps = {};
export default connect(null, mapDispatchToProps)(FontSizeSelector);
