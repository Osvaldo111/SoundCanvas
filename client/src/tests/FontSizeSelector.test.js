import React from "react";
import ReactDOM from "react-dom";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import FontSizeSelector from "../components/FontSizeSelector";
import { mount, shallow } from "enzyme";
//Redux
import { Provider } from "react-redux";
import store from "../reducers/store";
import { createSerializer } from "enzyme-to-json";
import { spy } from "sinon";
configure({ adapter: new Adapter() });

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Font size selector <FontSizeSelector />", () => {
  it("renders correctly", () => {
    const wrapper = mount(
      <Provider store={store}>
        <FontSizeSelector />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
