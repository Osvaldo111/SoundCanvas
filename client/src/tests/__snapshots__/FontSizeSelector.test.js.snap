// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Font size selector <FontSizeSelector /> renders correctly 1`] = `
<Connect(FontSizeSelector)>
  <div>
    <div>
      Font Size
    </div>
    <div
      className="ftSelConIn"
      onClick={[Function]}
    >
      <input
        onChange={[Function]}
        onKeyUp={[Function]}
        type="number"
        value="14"
      />
      <div
        className="ftSelConInIm"
      >
        <img
          alt=""
          src="arrowDown.svg"
        />
      </div>
    </div>
    <div
      className="ftSelCtnDrpDwn"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <ul
        onClick={[Function]}
      >
        <li
          data-size="6"
        >
          6
        </li>
        <li
          data-size="8"
        >
          8
        </li>
        <li
          data-size="10"
        >
          10
        </li>
        <li
          data-size="12"
        >
          12
        </li>
        <li
          data-size="14"
        >
          14
        </li>
        <li
          data-size="16"
        >
          16
        </li>
        <li
          data-size="18"
        >
          18
        </li>
        <li
          data-size="21"
        >
          21
        </li>
        <li
          data-size="24"
        >
          24
        </li>
        <li
          data-size="28"
        >
          28
        </li>
        <li
          data-size="32"
        >
          32
        </li>
        <li
          data-size="36"
        >
          36
        </li>
        <li
          data-size="42"
        >
          42
        </li>
        <li
          data-size="48"
        >
          48
        </li>
        <li
          data-size="56"
        >
          56
        </li>
      </ul>
    </div>
  </div>
</Connect(FontSizeSelector)>
`;
